doc/adr/0012-rewards.md

```diff
@@ -0,0 +1,44 @@
+# Rewards
+
+Date: 2022-06-01
+
+## Status
+
+Proposed
+
+## Context
+
+Facilitators get rewarded in DAG currency for participating in consensus rounds.
+We need to figure out the algorithm for minting/sending rewards to facilitators. Algorithm
+should take dynamic network speed into account to determine the size of rewards per node.
+
+## Decision
+
+1. Each node will track the time difference between the creation of local snapshots.
+We can either store the average or keep the specific number of meaningful diffs (e.g. 100 last diffs) and calculate the average on demand.
+The calculated average is **local network speed** and will be proposed to `GlobalSnapshot` consensus.
+
+Sample output: `3 min 15s`
+
+2. The snapshot speed should be rounded (e.g. to minutes) to increase the probability of selecting the value which occurs most often (dominant value). There are 2 possible results of finding dominant value: found or not.
```

Comment:
I'd say that depending on the standard deviation, we can round to different value (for example up to 10 minutes). Just throwing out an idea

Reply:
Agree strongly with this -- the only goal here presently is to give a rough approximation of time -- as opposed to dealing with scaling issues. In the long term, there is some dependency on the actual compute cost (which incentivizes nodes to spin up more instances in response to demand -- which requires some difficulty setting.) In the short term the only goal is to provide a rough measure of time for divisions, which means the time scale chosen can be relatively long, and has no requirement for capturing short variations. It can actually be much longer, but as mentioned earlier it's probably best to keep it medium because if it gets too great then nodes will have to wait to re-calculate that state -- or rely on externally downloaded non-synchronized state from another node.